src/archive/mod.rs

```diff
@@ -365,10 +405,27 @@ impl<'a> Archive<'a> {
 
             let name = member.raw_name();
             if name == INDEX_NAME {
+                if index_type == IndexType::BSD {
+                    return Err(Error::Malformed("SysV index occurs after BSD index".into()));
+                }
+                if index_type == IndexType::WindowsLinkerMember {
+                    return Err(Error::Malformed("More than two Windows Linker members".into()));
+                }
                 let data: &[u8] = buffer.pread_with(member.offset as usize, member.size())?;
-                index = Index::parse_sysv_index(data)?;
+                index = if index_type == IndexType::SysV {
+                    index_type = IndexType::WindowsLinkerMember;
+                    // second symbol index is Microsoft's extension of SysV format
+                    Index::parse_windows_linker_member(data)?
+                } else {
+                    index_type = IndexType::SysV;
+                    Index::parse_sysv_index(data)?
+                }
```

Comment:
I think this whole block would be clearer as:
```rust
match index_type {
    IndexType::None => { ... }
    IndexType::SysV => { ... }
    _ => return Err(...),
}
```

Reply:
I agree, it's clearer. Kept separate clauses for Windows/BSD to give more specific error messages.